### ISSUE TITLE
add number input widget

### ISSUE THREAD

**Issue by MasFlam** [dankamongmen/notcurses#1676]
We have `ncreader` which serves us well for textual input (albeit you have to filter `NCKEY_ENTER` yourself if you want single-line input). To accompany that, a numeric input widget would be useful for forms and the like. Should be a very simple widget, so I can do it in a couple of days or less, just opening an issue for it out of principle really.
The only design question I can think of is do we draw `+`/`-` / `^`/`v` on the widget? But that could and should probably be configurable. Oh, and a name for it. I have some painful experience with WinForms where it's called `NumericUpDown`; meanwhile Gtk calls it `GtkSpinButton`. So maybe `ncspin`...

**Comment by dankamongmen** [comment]
good call, i dig it. i'm wondering whether this is best done as a standalone widget, or whether `ncreader` ought be augmented with a `verify()` callback, invoked with each EGC input, that can reject or admit the EGC. perhaps independently, maybe there ought be a distinct `validate()` callback which can approve or disapprove of the text in its entirety. hrmmm.

**Comment by MasFlam** [comment]
> `ncreader` ought be augmented
Actually that's a good idea. But this kind of input in most places has a step and sometimes a minimum/maximum, and that would be problematic to implement with just both of the ideas you mentioned. We could make a widget that makes use of `ncreader` somehow though

**Comment by MasFlam** [comment]
I think making `ncspin` use `ncreader` is the right choice since you get to keep that direct input and you can implement value step, min/max etc.
Extending `ncreader` with validation callbacks can still be done separately anyways.
